### ISSUE TITLE
Unarchive module to support authorization when unarchiving from URL mentioned in src

### ISSUE THREAD

**Issue by AJ2090** [ansible/ansible-modules-core#4463]
##### ISSUE TYPE
- Feature Idea
##### COMPONENT NAME

Unarchive module
##### ANSIBLE VERSION

ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
##### OS / ENVIRONMENT

Ubuntu 16.04.1 LTS
##### SUMMARY

Currently to unarchive tarball from URL, we can use unarchive module as below.
- unarchive: src=https://example.com/example.zip dest=/usr/local/bin

But if authorization is required to access example.zip then module don't have such option. We can't unzip such files. Any option to give authorization credentials( as in get_url) will be helpful.
##### STEPS TO REPRODUCE

Unarchive module should have something like below to unzip authorized file from URL.
- unarchive: src=https://example.com/example.zip dest=/usr/local/bin force_basic_auth=yes url_username=testuser url_password=testpassword

```

```

##### ACTUAL RESULTS

```

```

**Comment by ansibot** [comment]
@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
Right, similar requests have come up and our current stance is that we don't want to duplicate all functionality of the `get_url` or `uri` module in the unarchive module. While it is convenient to have a single task that does everything under the kitchen sink, it will be a lot harder to support duplicate code.

@abadger @bcoca Maybe one of the Ansible devs can shed some light on how to handle these kinds of requests in the future ?

**Comment by dagwieers** [comment]
wontfix
